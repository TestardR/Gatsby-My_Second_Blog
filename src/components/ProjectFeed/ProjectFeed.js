// @flow strict
import React from 'react';
import moment from 'moment';
import { Link, withPrefix } from 'gatsby';
import type { Edges } from '../../types';
import styles from './ProjectFeed.module.scss';

type Props = {
  edges: Edges
};

const ProjectFeed = ({ edges }: Props) => {
  return (
    <div className={styles['project-feed']}>
      {edges.map(edge => (
        <div
          className={styles['project-feed__item']}
          key={edge.node.fields.slug}
        >
          <div className={styles['project-feed__item-meta']}>
            <h2 className={styles['project-feed__item-title']}>
              <Link
                className={styles['project-feed__item-title-link']}
                to={edge.node.fields.slug}
              >
                {edge.node.frontmatter.title}
              </Link>
            </h2>
            <div style={{display: 'flex'}}>
              <time
                className={styles['project-feed__item-meta-time']}
                dateTime={moment(edge.node.frontmatter.date).format(
                  'MMMM D, YYYY'
                )}
              >
                {moment(edge.node.frontmatter.date).format('YYYY')}
              </time>
              <span className={styles['project-feed__item-meta-divider']} />

              <span className={styles['project-feed__item-meta-time']}>{edge.node.frontmatter.time}</span>
            </div>
          </div>
          <div className={styles['project-feed__item-image']}>
            <Link
              className={styles['project-feed__item-image-link']}
              to={edge.node.fields.slug}
            >
              <img
                src={withPrefix(edge.node.frontmatter.socialImage)}
                className={styles['author__photo']}
                width="100%"
                height="100%"
                alt={edge.node.frontmatter.title}
              />
              <div className={styles['project-feed__item-image-overlay']}>
                <div
                  className={styles['project-feed__item-image-overlay-message']}
                >
                  Explore
                </div>
              </div>
            </Link>
          </div>
        </div>
      ))}
    </div>
  );
};

export default ProjectFeed;
