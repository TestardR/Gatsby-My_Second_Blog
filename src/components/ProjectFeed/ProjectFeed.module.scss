@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.project-feed {
  &__item {
    @include margin-bottom(1.25);

    &:last-child {
      @include margin-bottom(0.5);
    }

    &-title {
      font-size: $typographic-base-font-size * 1.6875;
      @include line-height(1.5);
      @include margin-top(0);
      @include margin-bottom(0.5);

      &-link {
        color: $color-base;

        &:hover,
        &:focus {
          color: $color-base;
          border-bottom: 1px solid $color-base;
        }
      }
    }

    &-image {
      position: relative;
      &-link {
        color: $color-base;

        &:hover,
        &:focus {
          color: $color-base;
          border-bottom: 1px solid $color-base;
        }
      }

      &-overlay {
        position: absolute;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
        height: 100%;
        width: 100%;
        opacity: 0;
        transition: 0.5s ease-in;
        background-color: black;

        &-message {
          color:white;
          font-size: 2.5rem;
          font-weight: 600;
          position: absolute;
          top: 50%;
          left: 50%;
          transform: translate(-50%, -50%);
          text-align: center;
        }
      }

      &:hover {
        .project-feed__item-image-overlay {
          opacity: .7;
        }
      }
    }

    &-meta {
      &-time {
        font-size: $typographic-small-font-size;
        color: #cacaca;
        font-weight: 500;
        // text-transform: uppercase;
        font-size: 1rem;
        line-height: 2rem;
        display: block;
        margin-bottom: 1rem;
        margin-top: -1.25rem;
      }
    }

    &-readmore {
      font-size: $typographic-base-font-size;
      color: $color-primary;

      &:hover,
      &:focus {
        color: $color-primary;
        border-bottom: 1px solid $color-primary;
      }
    }
  }
}
