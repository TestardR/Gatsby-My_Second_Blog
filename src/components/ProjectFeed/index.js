// @flow strict
export { default } from './ProjectFeed';
