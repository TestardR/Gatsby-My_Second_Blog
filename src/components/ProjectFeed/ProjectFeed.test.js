// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import ProjectFeed from './ProjectFeed';

describe('ProjectFeed', () => {
  const props = {
    edges: [
      {
        node: {
          fields: {
            slug: '/test_0'
          },
          frontmatter: {
            date: '2016-09-01',
            description: 'test_0',
            title: 'test_0',
            socialImage: '/test_0'
          }
        }
      }
    ]
  };

  it('renders correctly', () => {
    const tree = renderer.create(<ProjectFeed {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
