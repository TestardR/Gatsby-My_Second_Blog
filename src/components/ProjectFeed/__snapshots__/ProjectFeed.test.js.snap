// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectFeed renders correctly 1`] = `
<div
  className="project-feed"
>
  <div
    className="project-feed__item"
  >
    <div
      className="project-feed__item-meta"
    >
      <h2
        className="project-feed__item-title"
      >
        <a
          className="project-feed__item-title-link"
          href="/test_0"
        >
          test_0
        </a>
      </h2>
      <time
        className="project-feed__item-meta-time"
        dateTime="September 1, 2016"
      >
        September 2016
      </time>
    </div>
    <div
      className="project-feed__item-image"
    >
      <a
        className="project-feed__item-image-link"
        href="/test_0"
      >
        <img
          alt="test_0"
          className="author__photo"
          height="300px"
          src="/test_0"
          width="600px"
        />
        <div
          className="project-feed__item-image-overlay"
        >
          <div
            className="project-feed__item-image-overlay-message"
          >
            Explore
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
